File: solana/ts/src/circle/messageTransmitter/MessageTransmitterConfig.ts
import { PublicKey } from "@solana/web3.js";

export class MessageTransmitterConfig {
    owner: PublicKey;
    pendingOwner: PublicKey;
    attesterManager: PublicKey;
    pauser: PublicKey;
    paused: boolean;
    localDomain: number;
    version: number;
    signatureThreshold: number;
    enabledAttesters: Array<Array<number>>;
    maxMessageBodySize: bigint;
    nextAvailableNonce: bigint;
    authorityBump: number;

    constructor(
        owner: PublicKey,
        pendingOwner: PublicKey,
        attesterManager: PublicKey,
        pauser: PublicKey,
        paused: boolean,
        localDomain: number,
        version: number,
        signatureThreshold: number,
        enabledAttesters: Array<Array<number>>,
        maxMessageBodySize: bigint,
        nextAvailableNonce: bigint,
        authorityBump: number,
    ) {
        this.owner = owner;
        this.pendingOwner = pendingOwner;
        this.attesterManager = attesterManager;
        this.pauser = pauser;
        this.paused = paused;
        this.localDomain = localDomain;
        this.version = version;
        this.signatureThreshold = signatureThreshold;
        this.enabledAttesters = enabledAttesters;
        this.maxMessageBodySize = maxMessageBodySize;
        this.nextAvailableNonce = nextAvailableNonce;
        this.authorityBump = authorityBump;
    }

    static address(programId: PublicKey) {
        return PublicKey.findProgramAddressSync([Buffer.from("message_transmitter")], programId)[0];
    }
}

File: solana/modules/wormhole-cctp/src/cctp/message_transmitter_program/state/message_transmitter_config.rs
use anchor_lang::prelude::*;

#[derive(Debug, AnchorSerialize, AnchorDeserialize, Clone)]
pub struct MessageTransmitterConfig {
    pub owner: Pubkey,
    pub pending_owner: Pubkey,
    pub attester_manager: Pubkey,
    pub pauser: Pubkey,
    pub paused: bool,
    pub local_domain: u32,
    pub version: u32,
    pub signature_threshold: u32,
    pub enabled_attesters: Vec<[u8; 32]>,
    pub max_message_body_size: u64,
    pub next_available_nonce: u64,
}

impl anchor_lang::Discriminator for MessageTransmitterConfig {
    const DISCRIMINATOR: [u8; 8] = [71, 40, 180, 142, 19, 203, 35, 252];
}

impl Owner for MessageTransmitterConfig {
    fn owner() -> Pubkey {
        crate::cctp::message_transmitter_program::ID
    }
}
